compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Optics.hs

```diff
@@ -57,6 +59,14 @@ _PRSelfModule modName = prism (Qualified PRSelf modName) $ \case
   Qualified PRSelf modName' x | modName' == modName -> Right x
   q -> Left q
 
+_TypeSplitApp :: Iso' Type (Type, [Type])
+_TypeSplitApp = iso (splitOut []) joinIn
+  where
+    splitOut argsSoFar (TApp func arg) =
+      splitOut (arg:argsSoFar) func
+    splitOut argsSoFar func = (func, argsSoFar)
```

Comment:
This already exists as [`DA.Daml.LF.Ast.Util._TApps`](https://github.com/digital-asset/daml/blob/main/compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Util.hs#L235-L236)